### PR TITLE
Signal samples added to the analysis modes

### PR DESCRIPTION
**Pull request by jec429**
Signal samples added to analysis.cpp with the mode -m signal.

### CODE REVIEW

File: bin/analysis.cpp [hunk 1]
```diff
@@ -115,6 +115,8 @@ class chainNames
 			TTchainNames.push_back(dataTag + "_RunC");
 			TTchainNames.push_back(dataTag + "_RunD_v3");
 			TTchainNames.push_back(dataTag + "_RunD_v4");
+		} if(mode.find("WRto")!=std::string::npos){
```
Comment (shervin86) [inline bin/analysis.cpp]: Why mode is supposed to be "WRto...." instead of simply "signal"?

File: bin/analysis.cpp [hunk 2]
```diff
@@ -145,6 +147,10 @@ class chainNames
 
 	bool checkValidMode(std::string mode)
 	{
+	        if(mode.find("WRto")!=std::string::npos) {
```
Comment (shervin86) [inline bin/analysis.cpp]: std::string::npos can be replased by _ENDSTRING defined at the beginning of the file

File: bin/analysis.cpp [hunk 3]
```diff
@@ -237,6 +236,20 @@ int main(int ac, char* av[])
 		return 1;
 	}
 
+
+	std::cout << "[INFO] Selected modes: \n";
+	unsigned int msize = modes.size();
+	modes.erase( std::remove( modes.begin(), modes.end(), "signal" ), modes.end() );
+	if(modes.size() != msize){
```
Comment (shervin86) [inline bin/analysis.cpp]: Here you are adding all the signal samples, why this cannot be done at line 118 if mode==signal?

File: bin/analysis.cpp [hunk 4]
```diff
@@ -516,17 +529,16 @@ int main(int ac, char* av[])
...
-				// fill the tree with the normalization, parameters, and errors
-				tf1->Fill();
-			}
 			if(!run_toys)
 				break;
 		}
-		// only write the fitted branch for the modes that make sense (ttbar, DY, and data)
-		if(mode == "TT" || mode == "DY") // add the other modes later
-			tf1->Write();
+		// always write the fitted branch
```
Comment (shervin86) [inline bin/analysis.cpp]: fit are supposed to be done only on DY and TTBar, am I missing something trivial?

**Comment by shervin86** [comment]
I don't understand the logic. see comments inline.
Please use the script formatCode.sh bin/analysis.cpp before making a commit

File: bin/analysis.cpp (hunk 4 above)
Comment (jec429) [inline bin/analysis.cpp]: Peter added new branches to tf1 that should be saved for all the samples, i.e. the event count in the mass windows.

File: bin/analysis.cpp (hunk 3 above)
Comment (jec429) [inline bin/analysis.cpp]: This is to keep all the signal samples separate. If the modes vector is expanded before the getChainNames call, it is easier to process each sample at a time.

File: bin/analysis.cpp (hunk 2 above)
Comment (jec429) [inline bin/analysis.cpp]: Will change this.

File: bin/analysis.cpp (hunk 1 above)
Comment (jec429) [inline bin/analysis.cpp]: The vector of modes was expanded before the getChainNames call and thus the modes now correspond to the signal names.

File: bin/analysis.cpp (hunk 4 above)
Comment (shervin86) [inline bin/analysis.cpp]: Ok, then I'm fine with this change.

File: bin/analysis.cpp [hunk 5]
```diff
@@ -237,6 +237,20 @@ int main(int ac, char* av[])
 		return 1;
 	}
 
+
+	std::cout << "[INFO] Selected modes: \n";
+	unsigned int msize = modes.size();
+	modes.erase( std::remove( modes.begin(), modes.end(), "signal" ), modes.end() );
+	if(modes.size() != msize) {
```
Comment (shervin86) [inline bin/analysis.cpp]: When you say that modes are expanded before getChainNames you mean here, right?
Now I understand the logic of this. It seems to work, but it's very nasty. 
I'm wondering if we should define the modes in the datasets.dat as additional column, like:
datasetName ... ... ... mode
DYToEE        ... ... ... DY
and when you use "signal" as mode, then analysis.cpp looks into dataset.dat and make the push_back according to the samples defined there.
I would be nice if you can open an additional issue for this to be fixed in the next future.

**Comment by shervin86** [comment]
Fix #30

**Pull request merged**